hub/core/meta/tensor_meta.py

```diff
@@ -128,11 +132,19 @@ def update_with_sample(self, array: np.ndarray):
             self._update_shape_interval(shape)
 
     def _update_shape_interval(self, shape: Tuple[int, ...]):
+        if self.length <= 0:
+            self.min_shape = list(shape)
+            self.max_shape = list(shape)
         for i, dim in enumerate(shape):
             self.min_shape[i] = min(dim, self.min_shape[i])
             self.max_shape[i] = max(dim, self.max_shape[i])
 
 
+def _set_compression(htype_overwrite: dict, compression: str) -> dict:
+    htype_overwrite.update({"compression": compression})
+    return _set_compression
```

Comment:
it looks like this function is useless if you remove `compression` parameter (which means it will be contained in `**kwargs`). 

please remove this function after doing that

Reply:
This function was added to overcome a method that removes keys with None values so that compression can be None